clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11066,12 +11066,15 @@ def err_ext_int_max_size : Error<"%select{signed|unsigned}0 _ExtInt of bit "
 def err_esimd_glob_cant_init : Error<
   "SYCL explicit SIMD does not permit private global variable to have an initializer">;
 
+def err_sycl_kernel_name_missing : Error<"kernel name is missing">;
 def err_sycl_kernel_incorrectly_named : Error<
-  "kernel %select{name is missing"
-  "|needs to have a globally-visible name"
-  "|name is invalid. Unscoped enum requires fixed underlying type"
-  "|name cannot be a type in the \"std\" namespace"
-  "}0">;
+  "invalid type %0 used in kernel name. "
```

Comment:
you likely shouldn't change these diagnostics like this, diagnostics typically don't have periods/sentences/etc.

Reply:
I don't see this diagnostic used anywhere below: "invalid type %0 used in kernel name. "

Reply:
I find clang documentation about diagnostic subsystem quite helpful. It gives "best practices" for diagnostic message writing https://clang.llvm.org/docs/InternalsManual.html#the-format-string, as well as describes how diagnostics are invoked - https://clang.llvm.org/docs/InternalsManual.html#producing-the-diagnostic .

Reply:
@srividya-sundaram, the diagnostic is invoked using err_sycl_kernel_incorrectly_named. The diagnostic is 2 sentences. The call `QualType(Tag/ED->getTypeForDecl(), 0` is to pass the type (to `%0`) to emit the type in diagnostic.

Reply:
@erichkeane I agree. I just wasn't sure how to go about adding the problematic type as requested. Mariya's suggestion in a comment shows a better way. I'll make the change and upload a new patch.